src/types.ts

```diff
@@ -141,43 +205,181 @@ export type ManifestWorkflowSchema = {
 
 export type ManifestWorkflowsSchema = { [key: string]: ManifestWorkflowSchema };
 
-export type ManifestCustomTypeSchema = ParameterDefinition;
+export type ManifestMetadataSchema = {
+  "major_version"?: number;
+  "minor_version"?: number;
+};
 
-export type ManifestCustomTypesSchema = {
-  [key: string]: ManifestCustomTypeSchema;
+// Features
+export type ManifestBotUserSchema = {
+  "display_name": string;
+  "always_online"?: boolean;
+};
+
+// TODO: Find way to share these defaults
+type AppHomeMessagesTab = {
+  /** @default true */
+  messagesTabEnabled?: true;
+  /** @default true */
+  messagesTabReadOnlyEnabled?: boolean;
+} | {
+  /** @default true */
+  messagesTabEnabled: false;
+  /** @default true */
+  messagesTabReadOnlyEnabled: false;
+};
+
+export type ManifestAppHomeSchema = AppHomeMessagesTab & {
+  "home_tab_enabled"?: boolean;
+};
+
+export type SlackManifestFeaturesAppHome = ManifestAppHomeSchema;
+
+export type ManifestShortcutSchema = {
+  name: string;
+  type: "message" | "global";
+  "callback_id": string;
+  description: string;
+};
+
+export type ManifestShortcutsSchema = [
+  ManifestShortcutSchema,
+  ...ManifestShortcutSchema[],
+];
```

Comment:
Yes, but this notation is specifically for the array requiring at least 1 `ManifestShortcutSchema` type in it.  This is also how it's used in the [`webapp`](https://slack-github.com/slack/webapp/blob/master/js/modern/generated/types/api/schemas/app-manifests/v1/features/shortcuts.ts)

Reply:
Ohh, that makes perfect sense! Since we do this a lot, maybe a new Utility Type that indicates what it's doing would help? I.e.
```ts
type PopulatedArray<T> = [T, ...T[]];

export type ManifestShortcutsSchema = PopulatedArray<ManifestShortcutSchema>;
```